.github/workflows/agw-coverage.yml

```diff
@@ -0,0 +1,242 @@
+              - 'orc8r/**'
+              - 'lte/**'
+              - '.bazelrc'
+              - 'WORKSPACE.bazel'
+              - 'bazel/**'
+      - name: Save should_not_skip output
+        if: always()
+        run: |
+          mkdir -p ./pr
+          echo -n ${{ steps.changes.outputs.filesChanged == 'false' }} > ./pr/skipped
+      - uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # pin@v3
+        if: always()
+        with:
+          name: pr
+          path: pr/
+
+  c-cpp-codecov:
+    needs: path_filter
+    if: ${{ needs.path_filter.outputs.should_not_skip == 'true' }}
+    name: C / C++ code coverage
+    runs-on: ubuntu-latest
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # pin@v3.1.0
+      - name: Maximize build space
+        uses: ./.github/workflows/composite/maximize-build-space
+      - name: Setup Bazel Base Image
+        uses: addnab/docker-run-action@4f65fabd2431ebc8d299f8e5a018d79a769ae185 # pin@v3
+        with:
+          image: ${{ env.BAZEL_BASE_IMAGE }}
+          options: --pull always
+          # Run a simple echo command to pull down the image. This makes it a bit more clear how much time is spent on building Magma and not pulling down the image.
+          run: |
+            echo "Pulled the bazel base image!"
+            bazel # pull down bazel, if bazel download fails we can fail before we do all the lengthy work below
+      - name: Run C/C++ coverage with Bazel
+        if: always()
+        id: bazel-cc-codecoverage
+        uses: addnab/docker-run-action@4f65fabd2431ebc8d299f8e5a018d79a769ae185 # pin@v3
+        with:
+          image: ${{ env.BAZEL_BASE_IMAGE }}
+          # TODO: Remove work-around mount of Github workspace to /magma (https://github.com/addnab/docker-run-action/issues/11)
+          options: -v ${{ github.workspace }}:/workspaces/magma/ -v ${{ github.workspace }}/lte/gateway/configs:/etc/magma
+          run: |
+            cd $MAGMA_ROOT
+            bazel/scripts/remote_cache_bazelrc_setup.sh "${{ env.CACHE_KEY }}" "${{ env.REMOTE_DOWNLOAD_OPTIMIZATION_C_CPP }}" "${{ secrets.BAZEL_REMOTE_PASSWORD }}"
+            # Collecting coverage with Bazel can be slow. We can follow this thread to see if this can be improved: https://github.com/bazelbuild/bazel/issues/8178
+            # Coverage in bazel is flaky for remote caches - the flags below are helping. See GH13026 for details.
+            bazel coverage \
+              --profile=Bazel_test_cc_coverage_profile \
+              --experimental_split_coverage_postprocessing --experimental_fetch_all_coverage_outputs --remote_download_outputs=all \
+              //orc8r/gateway/c/...:* //lte/gateway/c/...:*
+            # copy out coverage information into magma so that it's accessible from the CI node
+            cp bazel-out/_coverage/_coverage_report.dat $MAGMA_ROOT
```

Comment:
This should probably be omitted in favour of a specification of `directory` in the codecov upload step.

Reply:
The file needs to be copied out of the folder in this step. Else it is deleted when the docker container is stopped.

Reply:
_Out of scope, but a general question:_
All important stuff in this workflow happens inside the bazel container, right?
Why not using Github build-ins to run this in a container than?
https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container

Reply:
Quick look at this: seems like it could work here, maybe also in the `bazel.yml` workflow. I have not tested the mounting etc. 

Would require some restructuring to be used in the other workflows. But the only advantage I see would be to use less external actions.

Reply:
Created ticket to investigate in the future. https://github.com/magma/magma/issues/14274

Reply:
Thanks! :pray: 
Security would also be an argument for using Github build-in functionality, of course.